### ISSUE TITLE
kops breaks when AWS_SDK_LOAD_CONFIG=1 is set

### ISSUE THREAD

**Issue by iMartyn** [kubernetes/kops#4484]
1. What `kops` version are you running? The command `kops version`, will display
 this information.
`master` currently at `8d5da313a`
2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.
n/a but 1.8.4

3. What cloud provider are you using?
aws

4. What commands did you run?  What is the simplest way to reproduce this issue?
`AWS_SDK_LOAD_CONFIG=1 kops-me create cluster --cloud=aws --dns public --networking flannel --zones=eu-west-1a,eu-west-1b,eu-west-1c --node-count=3 --master-zones=eu-west-1a,eu-west-1b,eu-west-1c --api-loadbalancer-type=public --state=s3://martyn-deletable --kubernetes-version 1.8.4 never-used.whatever.io --bastion=true --topology=private --target terraform`

5. What happened after the commands executed?
```
I0221 16:36:30.399583   31327 create_cluster.go:1316] Using SSH public key: /home/ranyardm/.ssh/id_rsa.pub
I0221 16:36:32.065324   31327 subnets.go:184] Assigned CIDR 172.20.32.0/19 to subnet eu-west-1a
I0221 16:36:32.065378   31327 subnets.go:184] Assigned CIDR 172.20.64.0/19 to subnet eu-west-1b
I0221 16:36:32.065399   31327 subnets.go:184] Assigned CIDR 172.20.96.0/19 to subnet eu-west-1c
I0221 16:36:32.065422   31327 subnets.go:198] Assigned CIDR 172.20.0.0/22 to subnet utility-eu-west-1a
I0221 16:36:32.065443   31327 subnets.go:198] Assigned CIDR 172.20.4.0/22 to subnet utility-eu-west-1b
I0221 16:36:32.065464   31327 subnets.go:198] Assigned CIDR 172.20.8.0/22 to subnet utility-eu-west-1c

error determining default DNS zone: error querying zones: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
```
6. What did you expect to happen?
More output and then
```Terraform output has been placed into out/terraform```

7. Please provide your cluster manifest. Execute
Easily recreated as the create command was provided.

8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.
```I0221 16:51:55.641434   32186 create_cluster.go:1316] Using SSH public key: /home/ranyardm/.ssh/id_rsa.pub
I0221 16:51:56.191948   32186 s3context.go:172] Found bucket "martyn-deletable" in region "eu-west-1"
I0221 16:51:56.192461   32186 s3fs.go:198] Reading file "s3://martyn-deletable/never-used.whatever.io/config"
I0221 16:51:56.370700   32186 channel.go:93] resolving "stable" against default channel location "https://raw.githubusercontent.com/kubernetes/kops/master/channels/"
I0221 16:51:56.370850   32186 channel.go:98] Loading channel from "https://raw.githubusercontent.com/kubernetes/kops/master/channels/stable"
I0221 16:51:56.370919   32186 context.go:145] Performing HTTP request: GET https://raw.githubusercontent.com/kubernetes/kops/master/channels/stable
I0221 16:51:56.454440   32186 channel.go:107] Channel contents: spec:
  images:
    # We put the "legacy" version first, for kops versions that don't support versions ( < 1.5.0 )
    - name: kope.io/k8s-1.4-debian-jessie-amd64-hvm-ebs-2017-07-28
      providerID: aws
      kubernetesVersion: ">=1.4.0 <1.5.0"
    - name: kope.io/k8s-1.5-debian-jessie-amd64-hvm-ebs-2018-01-14
      providerID: aws
      kubernetesVersion: ">=1.5.0 <1.6.0"
    - name: kope.io/k8s-1.6-debian-jessie-amd64-hvm-ebs-2018-01-14
      providerID: aws
      kubernetesVersion: ">=1.6.0 <1.7.0"
    - name: kope.io/k8s-1.7-debian-jessie-amd64-hvm-ebs-2018-01-14
      providerID: aws
      kubernetesVersion: ">=1.7.0 <1.8.0"
    - name: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-01-14
      providerID: aws
      kubernetesVersion: ">=1.8.0 <1.9.0"
    - name: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-01-14
      providerID: aws
      kubernetesVersion: ">=1.9.0 <1.10.0"
    # Need stretch as default in 1.10 (for nvme)
    # BUT... this is causing the submit queue to block, so back to jessie temporarily: https://github.com/kubernetes/kubernetes/issues/56763
    - name: kope.io/k8s-1.8-debian-jessie-amd64-hvm-ebs-2018-01-14
      providerID: aws
      kubernetesVersion: ">=1.10.0"
    - providerID: gce
      name: "cos-cloud/cos-stable-60-9592-90-0"
  cluster:
    kubernetesVersion: v1.5.8
    networking:
      kubenet: {}
  kubernetesVersions:
  - range: ">=1.8.0"
    recommendedVersion: 1.8.6
    requiredVersion: 1.8.0
  - range: ">=1.7.0"
    recommendedVersion: 1.7.11
    requiredVersion: 1.7.0
  - range: ">=1.6.0"
    recommendedVersion: 1.6.13
    requiredVersion: 1.6.0
  - range: ">=1.5.0"
    recommendedVersion: 1.5.8
    requiredVersion: 1.5.1
  - range: "<1.5.0"
    recommendedVersion: 1.4.12
    requiredVersion: 1.4.2
  kopsVersions:
  - range: ">=1.8.0-alpha.1"
    recommendedVersion: 1.8.0
    #requiredVersion: 1.8.0
    kubernetesVersion: 1.8.6
  - range: ">=1.7.0-alpha.1"
    recommendedVersion: 1.7.1
    #requiredVersion: 1.7.0
    kubernetesVersion: 1.7.11
  - range: ">=1.6.0-alpha.1"
    #recommendedVersion: 1.6.0
    #requiredVersion: 1.6.0
    kubernetesVersion: 1.6.13
  - range: ">=1.5.0-alpha1"
    recommendedVersion: 1.5.1
    #requiredVersion: 1.5.1
    kubernetesVersion: 1.5.8
  - range: "<1.5.0"
    recommendedVersion: 1.4.4
    #requiredVersion: 1.4.4
    kubernetesVersion: 1.4.12
I0221 16:51:56.455171   32186 create_cluster.go:877] networking mode=flannel => {"flannel":{"backend":"vxlan"}}
I0221 16:51:56.455265   32186 aws_utils.go:39] Querying EC2 for all valid regions
I0221 16:51:57.111126   32186 aws_cloud.go:966] Querying EC2 for all valid zones in region "eu-west-1"
I0221 16:51:57.111473   32186 request_logger.go:45] AWS request: ec2/DescribeAvailabilityZones
I0221 16:51:57.285342   32186 subnets.go:184] Assigned CIDR 172.20.32.0/19 to subnet eu-west-1a
I0221 16:51:57.285399   32186 subnets.go:184] Assigned CIDR 172.20.64.0/19 to subnet eu-west-1b
I0221 16:51:57.285418   32186 subnets.go:184] Assigned CIDR 172.20.96.0/19 to subnet eu-west-1c
I0221 16:51:57.285436   32186 subnets.go:198] Assigned CIDR 172.20.0.0/22 to subnet utility-eu-west-1a
I0221 16:51:57.285469   32186 subnets.go:198] Assigned CIDR 172.20.4.0/22 to subnet utility-eu-west-1b
I0221 16:51:57.285496   32186 subnets.go:198] Assigned CIDR 172.20.8.0/22 to subnet utility-eu-west-1c
I0221 16:51:57.285518   32186 defaults.go:208] Not setting up Proxy Excludes
I0221 16:51:57.289943   32186 populate_cluster_spec.go:380] Defaulted KubeControllerManager.ClusterCIDR to 100.96.0.0/11
I0221 16:51:57.289983   32186 populate_cluster_spec.go:387] Defaulted ServiceClusterIPRange to 100.64.0.0/13
I0221 16:51:57.290016   32186 aws_cloud.go:966] Querying EC2 for all valid zones in region "eu-west-1"
I0221 16:51:57.290284   32186 request_logger.go:45] AWS request: ec2/DescribeAvailabilityZones
I0221 16:51:57.330094   32186 subnets.go:49] All subnets have CIDRs; skipping asssignment logic
I0221 16:51:57.330149   32186 defaults.go:208] Not setting up Proxy Excludes
I0221 16:51:57.330254   32186 aws_cloud.go:966] Querying EC2 for all valid zones in region "eu-west-1"
I0221 16:51:57.330619   32186 request_logger.go:45] AWS request: ec2/DescribeAvailabilityZones
I0221 16:51:57.371326   32186 utils.go:151] Querying for all DNS zones to find match for "never-used.whatever.io"
I0221 16:51:57.371555   32186 route53.go:50] AWS request: route53 ListHostedZones

error determining default DNS zone: error querying zones: NoCredentialProviders: no valid providers in chain. Deprecated.
	For verbose messaging see aws.Config.CredentialsChainVerboseErrors```

9. Anything else do we need to know?
my `~/.aws/config` :
```[default]
region = eu-west-1```

**Comment by layer3switch** [comment]
It seems, the ```assumeRole``` for default profile or the account doesn't allow ```Route53``` ```ListHostedZones``` call permission. Can you perform ```aws route53 list-hosted-zones``` command and see if it works first?

FYI: if aws api requires MFA enforcement, there is open issue for this https://github.com/kubernetes/kops/issues/226

**Comment by iMartyn** [comment]
Given that my aws cli user is an aws admin account and this is not an assumeRole account, but a normal admin iam aws account, I don't see that that is the cause.  Add to that that without the AWS_SDK_LOAD_CONFIG=1 I'm using the same account with the same permissions, I'm pretty sure it's not that.

I cannot run that command until Wednesday now.

**Comment by iMartyn** [comment]
@layer3switch nope, `AWS_SDK_LOAD_CONFIG=1 aws route53 list-hosted-zones` works fine, that's not the problem.
@chrislovecnm I really think that this issue should be included in the 1.9.0 backlog

**Comment by chrislovecnm** [comment]
The aws cli and the go api work in very different mysterious ways ;(

We can include this in the milestone, but we are about to cut the alpha.

Do we have a fix for this?

**Comment by chrislovecnm** [comment]
Is this broken in kops 1.8.1?

**Comment by iMartyn** [comment]
Yeah the sdk and cli are not very consistent.

This works fine in 1.8.1 and for multi-account you kinda need the AWS_SDK_LOAD_CONFIG=1.  To release 1.9 without the fix will break many peoples' setups, so I think we need to find a fix.  I'm not aware of one yet, haven't tried to fix this issue, because I found this one fixing the other one and got wiped out.

**Comment by chrislovecnm** [comment]
Is there an issue open with the aws go api?

**Comment by chrislovecnm** [comment]
https://github.com/aws/aws-sdk-go/search?q=AWS_SDK_LOAD_CONFIG&type=Issues&utf8=✓

There are some bugs, not sure which version is fixed compared to our vendorred version.

**Comment by chrislovecnm** [comment]
Fixed in aws go api v1.12.72

**Comment by iMartyn** [comment]
Can confirm, issue no longer there in master.
